src/sage/rings/function_field/drinfeld_modules/finite_drinfeld_module.py

```diff
@@ -352,25 +555,10 @@ def frobenius_trace(self):
 
         ALGORITHM:
 
-            Let `A(T)` denote the Frobenius trace and `B(T)` denote the
-            Frobenius norm. We begin by computing `B(T)`, see docstring
-            of method :meth:`frobenius_norm` for details. The
-            characteristic polynomial of the Frobenius yields `t^{2n} -
-            \phi_A t^n + \phi_B = 0`, where `t^n` is the Frobenius
-            endomorphism. As `\phi_B` is now known, we can compute
-            `\phi_A = (t^{2n} + \phi_B) / t^n`. We get `A(T)` by
-            inverting this quantity, using the method
-            :meth:`sage.rings.function_fields.drinfeld_module.drinfeld_module.DrinfeldModule.invert`,
-            see its docstring for details.
+            We extract the coefficient of `X^{r-1}` from the
+            characteristic polynomial.
         """
-        self._check_rank_two()
-        # Notations from Schost-Musleh:
-        if self._frobenius_trace is None:
-            n = self._base.over(self._Fq).degree_over(self._Fq)
-            B = self.frobenius_norm()
-            t = self.ore_polring().gen()
-            self._frobenius_trace = self.invert(t**n + (self(B) // t**n))
-        return self._frobenius_trace
+        return -self.frobenius_charpoly().coefficients(sparse=False)[-2]
```

Comment:
Same remark as above. (Even more relevant, here, as a trace computation is really faster than that of a characteristic polynomial.)